gaia-header.js

```diff
@@ -145,13 +327,172 @@ module.exports = Component.register('gaia-header', {
    * @param  {Event} e
    * @private
    */
-  onActionButtonClick: function(e) {
-    var config = { detail: { type: this.getAttribute('action') } };
-    var actionEvent = new CustomEvent('action', config);
-    setTimeout(this.dispatchEvent.bind(this, actionEvent));
+  onActionButtonClick: function() {
+    debug('action button click');
+    var config = { detail: { type: this.action } };
+    var e = new CustomEvent('action', config);
+    setTimeout(() => this.dispatchEvent(e));
+  },
+
+  /**
+   * Get the margin required to center
+   * the title between surrounding buttons.
+   *
+   * We don't do any DOM writes in this function
+   * so that we can batch them together with
+   * other write for optimum performance.
+   *
+   * @param  {Object} title
+   * @return {Object}
+   * @private
+   */
+  getTitleMarginStart: function() {
+    var titleStart = this.titleStart;
+    var titleEnd = this.titleEnd;
+    debug('get title margin', titleStart, titleEnd);
+    return titleStart === titleEnd ? 0 : titleEnd - titleStart;
+  },
+
+  /**
+   * Get all the buttons (<a> & <button>)
+   * before the first <h1>.
+   *
+   * @return {Array}
+   * @private
+   */
+  getButtonsBeforeTitle: function() {
+    var els = [];
+
+    for (var i = 0; i < this.children.length; i++) {
```

Comment:
it's better if you cache this property; it's not a normal array here, and I think accessing length is O(n) (but I'm not sure, and n is very small here, so it's really a nit)